src/sdk/Viewport3d.ts

```diff
@@ -216,6 +216,9 @@ export class Viewport3d implements ViewportDelegate {
   }
 
   onKeyDown(e: KeyboardEvent) {
+    if (e.key === 'Shift') {
+      Viewport.viewport.contextMenu.holdingShift = true;
```

Comment:
Can we keep the `holdingShift` state in Viewport3d (or something higher level, and keep it together)? It seems odd to me that we have the listener here, but it's storing state somewhere lower.